src/Instrumentation/Laravel/_register.php

```diff
@@ -2,17 +2,23 @@
 
 declare(strict_types=1);
 
+use Nevay\SPI\ServiceLoader;
+use OpenTelemetry\API\Instrumentation\AutoInstrumentation\Instrumentation;
+use OpenTelemetry\Contrib\Instrumentation\Laravel\Hooks;
 use OpenTelemetry\Contrib\Instrumentation\Laravel\LaravelInstrumentation;
-use OpenTelemetry\SDK\Sdk;
 
-if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(LaravelInstrumentation::NAME) === true) {
-    return;
-}
+// In case the `tbachert/spi` plugin has not been allowed in composer allow-plugins (root-level).
+ServiceLoader::register(Instrumentation::class, LaravelInstrumentation::class);
 
-if (extension_loaded('opentelemetry') === false) {
-    trigger_error('The opentelemetry extension must be loaded in order to autoload the OpenTelemetry Laravel auto-instrumentation', E_USER_WARNING);
+ServiceLoader::register(Hooks\Hook::class, Hooks\Illuminate\Console\Command::class);
```

Comment:
Having mapped all these out here, I might as well have kept them in `LaravelInstrumentation` and just called them there without SPI.

Reply:
Thanks to the work from @Nevay (and [these instructions](https://github.com/Nevay/spi/issues/1#issuecomment-2198483721)), this has now been improved to de-dupe registration.